crates/fuel-core/src/service/adapters/fuel_gas_price_provider.rs

```diff
@@ -30,20 +30,27 @@ mod tests;
 #[derive(Debug)]
 /// Receives the next gas price algorithm via a shared `BlockGasPriceAlgo` instance
 pub struct FuelGasPriceProvider<A> {
+    // // Scale the gas price down by this factor, for e.g. Wei to Gwei
+    // scaling_factor: u64,
     algorithm: SharedGasPriceAlgo<A>,
 }
 
 impl<A> Clone for FuelGasPriceProvider<A> {
```

Comment:
do we need this impl if we just restrict `A: Clone` ?

Reply:
We don't really care if `A: Clone` though, this is more precise.